Model field with `chrono::DateTime<chrono::Local>` panic

Hi team,

I've used sqlx with the following model:

```
CREATE TABLE `user` (
  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
  `ctime` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `mtime` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  `is_deleted` tinyint(4) COLLATE utf8_general_ci NOT NULL DEFAULT 0,
  `email` varchar(255) COLLATE utf8_general_ci NOT NULL DEFAULT '',
  `name` varchar(255) COLLATE utf8_general_ci NOT NULL DEFAULT '',
  PRIMARY KEY (`id`),
  KEY `ix_mtime` (`mtime`)
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8 COLLATE=utf8_general_ci;
```

```
...
#[derive(FromRow, Serialize, Deserialize, Debug)]
pub struct User {
    pub id: SqlID,
    pub ctime: chrono::DateTime<Utc>,
    pub mtime: chrono::DateTime<Utc>,
    pub name: String,
    pub email: String,
}
...
let result = sqlx::query_as("SELECT * FROM user")
    .fetch_all(pool.get_ref()).await;
```

and it compiles with error below:

```
the trait `sqlx::Type<sqlx::MySql>` is not implemented for `chrono::DateTime<chrono::Local>`
```

If I changed to `chrono::DateTime<chrono::Utc>`, it works.

config in `Cargo.toml` likes below:

```
sqlx = { version = "0.4.2", features = [ "runtime-actix-native-tls", "mysql", "chrono" ] }
chrono =  { version = "0.4.15", features = ["serde"] }
```

Help!

me too!
```
error[E0277]: the trait bound `DateTime<Local>: types::Type<MySql>` is not satisfied
   --> C:\Users\10342\.cargo\registry\src\github.com-1ecc6299db9ec823\sqlx-core-0.5.5\src\any\type.rs:24:25
    |
24  |                         <$ty as crate::types::Type<crate::mysql::MySql>>::compatible(&ty)
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `types::Type<MySql>` is not implemented for `DateTime<Local>`
    | 
   ::: C:\Users\10342\.cargo\registry\src\github.com-1ecc6299db9ec823\sqlx-core-0.5.5\src\types\mod.rs:177:5
    |
177 |     fn compatible(ty: &DB::TypeInfo) -> bool {
    |     ---------------------------------------- required by `types::Type::compatible`
    | 
   ::: C:\Users\10342\.cargo\registry\src\github.com-1ecc6299db9ec823\sqlx-core-0.5.5\src\any\types.rs:127:1
    |
127 | impl_any_type!(chrono::DateTime<chrono::offset::Local>);
    | -------------------------------------------------------- in this macro invocation
    |
    = help: the following implementations were found:
              <DateTime<Local> as types::Type<any::database::Any>>
              <DateTime<Tz> as types::Type<Sqlite>>
              <DateTime<Tz> as types::Type<postgres::database::Postgres>>
              <DateTime<Utc> as types::Type<MySql>>
              <DateTime<Utc> as types::Type<any::database::Any>>
    = note: this error originates in the macro `impl_any_type` (in Nightly builds, run with -Z macro-backtrace for more info)

```

--- a/sqlx-core/src/mysql/types/chrono.rs
+++ b/sqlx-core/src/mysql/types/chrono.rs
@@ -1,7 +1,7 @@
 use std::convert::TryFrom;
 
 use bytes::Buf;
-use chrono::{DateTime, Datelike, NaiveDate, NaiveDateTime, NaiveTime, Timelike, Utc};
+use chrono::{DateTime, Datelike, Local, NaiveDate, NaiveDateTime, NaiveTime, Timelike, Utc};
 
 use crate::decode::Decode;
 use crate::encode::{Encode, IsNull};
@@ -21,17 +21,43 @@ impl Type<MySql> for DateTime<Utc> {
     }
 }
 
+/// Note: assumes the connection's `time_zone` is set to `+00:00` (UTC).
 impl Encode<'_, MySql> for DateTime<Utc> {
     fn encode_by_ref(&self, buf: &mut Vec<u8>) -> IsNull {
         Encode::<MySql>::encode(&self.naive_utc(), buf)
     }
 }
 
+/// Note: assumes the connection's `time_zone` is set to `+00:00` (UTC).
 impl<'r> Decode<'r, MySql> for DateTime<Utc> {
     fn decode(value: MySqlValueRef<'r>) -> Result<Self, BoxDynError> {
         let naive: NaiveDateTime = Decode::<MySql>::decode(value)?;
 
         Ok(DateTime::from_utc(naive, Utc))
+    }
+}
+
+impl Type<MySql> for DateTime<Local> {
+    fn type_info() -> MySqlTypeInfo {
+        MySqlTypeInfo::binary(ColumnType::Timestamp)
+    }
+
+    fn compatible(ty: &MySqlTypeInfo) -> bool {
+        matches!(ty.r#type, ColumnType::Datetime | ColumnType::Timestamp)
+    }
+}
+
+/// Note: assumes the connection's `time_zone` is set to `+00:00` (UTC).
+impl Encode<'_, MySql> for DateTime<Local> {
+    fn encode_by_ref(&self, buf: &mut Vec<u8>) -> IsNull {
+        Encode::<MySql>::encode(&self.naive_utc(), buf)
+    }
+}
+
+/// Note: assumes the connection's `time_zone` is set to `+00:00` (UTC).
+impl<'r> Decode<'r, MySql> for DateTime<Local> {
+    fn decode(value: MySqlValueRef<'r>) -> Result<Self, BoxDynError> {
+        Ok(<DateTime<Utc> as Decode<'r, MySql>>::decode(value)?.with_timezone(&Local))
     }
 }
 

